pages/devfiles/[id]/index.tsx

```diff
@@ -21,36 +24,32 @@ interface Path {
  * @param devfileText - text of devfile YAML, null when sample
  * @param devfileJSON -  json representation of devfile YAML, null when sample
  */
-const DevfilePage = ({
+const DevfilePage: React.FC<InferGetStaticPropsType<typeof getStaticProps>> = ({
   devfile,
   devfileYAML,
   devfileJSON,
   errors
 }: InferGetStaticPropsType<typeof getStaticProps>) => (
-  <div style={{ alignContent: 'center', minHeight: '100vh' }}>
+  <div className={styles.devfilePage}>
     <ErrorBanner errors={errors} />
-    {devfile.type === 'stack' ? (
-      <div>
-        <DevfilePageHeader devfileMetadata={devfileJSON.metadata} devfile={devfile} />
-        <DevfilePageProjects starterProjects={devfileJSON.starterProjects} />
-        <DevfilePageYAML devfileYAML={devfileYAML} />
-      </div>
-    ) : (
-      <DevfilePageHeader devfile={devfile} />
+    <DevfilePageHeader devfileMetadata={devfileJSON?.metadata} devfile={devfile} />
+    {devfileJSON?.starterProjects?.length && (
+      <DevfilePageProjects starterProjects={devfileJSON.starterProjects} />
     )}
+    {devfileYAML && <DevfilePageYAML devfileYAML={devfileYAML} />}
   </div>
 );
 
 export const getStaticProps: GetStaticProps = async (context) => {
-  const [devfiles, devfileErrors]: GetMetadataOfDevfiles = await getMetadataOfDevfiles();
+  const [devfiles, devfileErrors] = await getMetadataOfDevfiles();
 
   const devfile: Devfile = devfiles.find((devfile: Devfile) => {
     const id = context.params?.id as string;
     const [source, name] = id.split('+');
     return devfile.sourceRepo === source && devfile.name === name;
   })!;
 
-  const [devfileYAML, devfileJSON, yamlErrors]: GetDevfileYAML = await getDevfileYAML(devfile);
+  const [devfileYAML, devfileJSON, yamlErrors] = await getDevfileYAML(devfile);
```

Comment:
I mean it doesn't really matter but from what I have seen is that you do not want redundant types. Since we assign the return type in the function this becomes redundant. Also, if you hover over the variables intellisense gives you the types.